packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -476,4 +486,39 @@ export const useGridSelection = (
       }
     }
   }, [apiRef, isRowSelectable, isStateControlled]);
+
+  React.useEffect(() => {
+    const currentSelection = gridSelectionStateSelector(apiRef.current.state);
+
+    if (!checkboxSelection && disableMultipleSelection && currentSelection.length > 1) {
+      const { rows: currentPageRows } = getCurrentPageRows(apiRef.current.state, {
+        pagination,
+        paginationMode,
+      });
+
+      const currentPageRowsLookup = currentPageRows.reduce((acc, { id }) => {
+        acc[id] = true;
+        return acc;
+      }, {});
+
+      const firstSelectableRow = currentSelection.find((id) => {
+        let isSelectable = true;
+        if (typeof isRowSelectable === 'function') {
```

Comment:
```suggestion
        if (isRowSelectable) {
```

Like in line 479. I was looking for what are the other types of `isRowSelectable`